### DISCUSSION TITLE
Usage of capacitor-plugins with @ionic-core.

### DISCUSSION THREAD

**Discussion by johnborges** [ionic-team/capacitor#4221] [Q&A] [answered]
While looking through the migration docs, I noticed the new [recommendation](https://capacitorjs.com/docs/v3/updating/3-0#migrating-your-app-to-use-the-new-official-plugin-packages) is to include these plugins: `npm install @capacitor/app @capacitor/haptics @capacitor/keyboard` in Capacitor apps that also use the Ionic Framework. 

Is there an example somewhere that shows any usage of those plugins in `@ionic/core`? I'm trying to get a better understanding of what features in ionic make use of those plugins.

**Comment by jcesarmobile** [comment, marked as answer]
Well, ionic code is open source too, you just need to search for haptics and find
https://github.com/ionic-team/ionic-framework/blob/master/core/src/utils/native/haptic.ts

Search for keyboard and find 
https://github.com/ionic-team/ionic-framework/blob/master/core/src/utils/keyboard/keyboard.ts

App is harder to find, but it’s used for the back button handling, so if you search for back button you find
https://github.com/ionic-team/ionic-framework/blob/master/core/src/utils/hardware-back-button.ts
